frontend/public/components/instantiate-template.tsx

```diff
@@ -57,7 +58,7 @@ const TemplateInfo: React.FC<TemplateInfoProps> = ({ template }) => {
   const { description } = annotations;
   const displayName = annotations[ANNOTATIONS.displayName] || template.metadata.name;
   const iconClass = getTemplateIcon(template);
-  const imgURL = getImageForIconClass(iconClass);
+  const imgURL = iconClass ? getImageForIconClass(iconClass) : catalogImg;
```

Comment:
Is it a possibility that `iconClass` is defined but `getImageForIconClass(iconClass)` returns undefined?

Reply:
I think it is possible that `imgURL` could still be `undefined` after this line. We should probably either adjust `getImageForIconClass` function to accept a `fallback` arg, or better handle the case here. I lean towards the first option because it is reusable and will make this line simpler:

```suggestion
  const imgURL = getImageForIconClass(iconClass) ?? catalogImg;
```

Reply:
The above suggestion is incorrect because the `iconClass` if defined, but not resolvable to a logo, we still want to fallback to using the `iconClass` as a font icon.
The code certainly seems odd but it's legacy...

Reply:
@invincibleJai and @TheRealJon - As @christianvogt mentioned this is as expected. Because we have 3 different cases here. If the imgURL was undefined the Patternfly component will render the iconClass. I tried to keep the diff here as small as possible.

Checkout the images in the PR to check the different cases.

Is this acceptable for you for now? This PR would be merged when you remove your "needs a code change" lock. :smile:

Reply:
See also https://github.com/openshift/console/blob/629c40752976def0a63163def546035ad50a785f/frontend/public/components/instantiate-template.tsx#L70-L82